src/AngleSharp.Css/Extensions/MediaListExtensions.cs

```diff
@@ -40,7 +40,18 @@ public static Boolean Validate(this ICssMedium medium, IRenderDevice device)
                 return false;
             }
 
-            return !medium.IsInvalid(device) && !medium.Features.Any(m => m.Validate(device) == medium.IsInverse);
+            return !medium.IsInvalid(device) && !medium.Features.Any(m =>
+            {
+                try
+                {
+                    return m.Validate(device) == medium.IsInverse;
+                }
+                catch (InvalidOperationException x) when(x.Message.StartsWith("Unsupported unit"))
```

Comment:
```suggestion
                catch (InvalidOperationException ex) when (ex.Message.StartsWith("Unsupported unit"))
```

WidthFeatureValidator.cs:16 would be a better place for this hack.

Reply:
or maybe even in `AsPx`